src/System.Management.Automation/engine/parser/TypeResolver.cs

```diff
@@ -690,7 +690,8 @@ internal static class CoreTypes
                     { typeof(PSCredential),                                new[] { "pscredential" } },
                     { typeof(PSDefaultValueAttribute),                     new[] { "PSDefaultValue" } },
                     { typeof(PSListModifier),                              new[] { "pslistmodifier" } },
-                    { typeof(PSObject),                                    new[] { "psobject", "pscustomobject" } },
+                    { typeof(PSObject),                                    new[] { "psobject" } },
+                    { typeof(PSCustomObject),                              new[] { "pscustomobject" } },
```

Comment:
This is a breaking change. I realize it feels wrong, but it's been this way since V1, we can't really change it.